### DISCUSSION TITLE
Betrieb ohne Internet Router und evtl. ohne AccessPoint?

### DISCUSSION THREAD

**Discussion by haegar33** [hoylabs/OpenDTU-OnBattery#1080] [Q&A] [answered]
Hallo!

Der ESP32 baut ja offenbar wie bei vielen ESP Projekten nur zur Initialisierung einen Wifi AP auf um sich dann im "Normalbetrieb" auf einen externen AP zu verlassen. Ginge das auch anders? 

1. Interner AP bleibt aktiv, WR und andere Komponenten (i.e. VE-Bus..) kommunizieren direkt -> keine Kontrolle, kein MQTT möglich.
2. Es gibt im Haus einen externen AP aber keinen Router -> Kontrolle und Steuerung über WebGUI möglich, ebenso MQTT Nachrichten, natürlich keine Zeit und Datums Info über Time Server.

Die Idee wäre nur zum ersten Setup OpenDTU mit dem Internet zu verbinden und danach aber lokal zu bleiben. Welche Infos gehen dann bei einem Stromausfall verloren (außer die Uhrzeit)?

**Comment by spcqike** [comment]
So ganz verstehe ich dein setup nicht. 

die openDTU an sich braucht kein Internet. Nur einen NTP Server für den Fall, dass sie Neustarten oder stromlos wird. Ohne korrekter Systemzeit kann keine Verbindung zum wechselrichter aufgebaut werden. 

Alles andere kann im lokalen Netz erfolgen, sofern eins existiert. Bspw MQTT, ein MQTT Broker müsste auf einem lokalen Host laufen. 

Was du machen kannst, und ich früher auch hatte, einen raspberry als AP mit NTP und MQTT broker laufen lassen. Dann kannst du offline im lokalen Netz alles abbilden was du brauchst.

**Comment by haegar33** [comment]
Es geht um den Einsatz in einem Ferienhaus ohne Internet. Ja, vermutlich brauche ich einen seperaten Wifi AP (Raspberry, 2.ter ESP32 o.ä.) da vermutlich ein (OpenDTU) ESP32 überfordert wäre.
Das eine korrekte Systemzeit notwenig ist für die Verbindung zum WR notwendig ist wusste ich nicht. Wie groß darf hierbei die Abweichung sein? 
Idee ist im Ferienhaus eine erste Initialisierung mit einem AP/Router über mein Smartphone zu machen und dann das System weiter laufen zu lassen.

**Reply by schlimmchen** [reply]
Das Zeitsynchronisierungsproblem musst du lösen. Geht aber im Zweifel vom Smartphone aus (In der Web UI der OpenDTU-OnBattery über den Browser synchronisieren). Ansonsten versuch doch mal das Admin-AP Timeout auf Null zu setzen. Jedenfalls scheint 0 erlaubt zu sein. Bleibt der AP dann dauerhaft an? Dann kannst du auch anschließend immer mal wieder draufgucken und justieren.

**Reply by haegar33** [reply]
Habe inzwischen das FusionBoard mit vorinstalliertem OpenDTU getestet. Alles läuft, auch die Verbindung zu meinem Hoymiles 600 ;)
Den Admin-AP Timeout auf null gesetzt funktioniert solange der ESP32 kein anderen Wifi AP findet. Ist der wieder weg bleibt der interne AP activ, sehr gut! Ich werde natürlich die OpenDTUonBattery Firmware aufspielen aber erwarte das gleiche Verhalten?

1.Ausbauschritt ist eine "einfache" Nulleinspeisung mit einem Tasmota IR Lesekopf. Auch hier würde ich hoffen das der "Tasmota ESP" direkt als httpclient vom FusionBoard ansprechbar ist - ohne externen Wifi AP?

**Reply by spcqike** [reply, marked as answer]
> Auch hier würde ich hoffen das der "Tasmota ESP" direkt als httpclient vom FusionBoard ansprechbar ist - ohne externen Wifi AP?

gehe ich erstmal von aus. Hab es zwar selber so nie getestet oder betrieben, aber warum sollte es sich anders verhalten? 

Solange das Fusion Board den AP offen lässt und der tasmota sich mit dem AP verbindet, eine IP aus 192.168.4.x (ich glaub ich würd da ne feste vergeben an deiner Stelle, ich weiß nicht wie openDTU da den dhcp lease behandelt und hält) und so auch erreichbar ist …. Sollte ein http Request klappen. 

MQTT geht so natürlich nicht.

**Comment by haegar33** [comment]
Nachtrag zur Frage: Gestern ein 2.ESP32 Board mit Tasmota geflasht und  dort das FusionBoard als Wifi AP eingetragen.
In openDTUonBattery ein (Pseudo)SmartMeter mit HTTP/JSON unter der IP 192.168.4.2 eingetragen. Die Verbindung funktioniert und HTTP GET commands liefern JSON Strings auf dem FusionBoard ;)

**Reply by spcqike** [reply]
Super dass es klappt. Danke für die Rückmeldung. 

Wie gesagt, im Falle eines Stromausfalls musst du dir halt Gedanken über NTP Synchronisation machen. Sonst erreicht die DTU den wechselrichter nicht mehr.
